File: app/src/main/java/com/thejoyrun/aptpreferences/MainActivity.java
package com.thejoyrun.aptpreferences;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.widget.TextView;

import com.thejoyrun.aptpreferences.preferences.Settings;
import com.thejoyrun.aptpreferences.preferences.SettingsPreferences;

import java.util.Date;

public class MainActivity extends AppCompatActivity {


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        SettingsPreferences settingsPreference = SettingsPreferences.get("name");
        StringBuilder stringBuilder = new StringBuilder();
        stringBuilder.append("last value firstUse: ").append(settingsPreference.getUseLanguage()).append('\n');
        settingsPreference.setUseLanguage(new Date().toString());
        TextView textView = (TextView) findViewById(R.id.text);
        textView.setText(stringBuilder.toString());
//        SettingsPreferences2.get()

//        settingsPreference.getRun().getVoiceName();
        // 普通类型保存
        settingsPreference.setUseLanguage("zh");
        settingsPreference.setLastOpenAppTimeMillis(System.currentTimeMillis());
        // 对象类型保存
        Settings.LoginUser loginUser = new Settings.LoginUser();
        loginUser.setUsername("username");
        loginUser.setPassword("password");
        settingsPreference.setLoginUser(loginUser);
        // 对象类型带 @AptField(preferences = true) 注解的保存，相当于把 push相关的放在一个分类
        settingsPreference.getPush().setOpenPush(true);


        // 获取
        String useLanguage = settingsPreference.getUseLanguage();
        Settings.LoginUser loginUser1 = settingsPreference.getLoginUser();
        boolean openPush = settingsPreference.getPush().isOpenPush();

    }

}

File: app-kotlin/src/main/java/com/thejoyrun/app_kotlin/MainActivity.kt
package com.thejoyrun.app_kotlin

import android.app.Activity
import android.os.Bundle
import com.thejoyrun.app_kotlin.preferences.Settings
import com.thejoyrun.app_kotlin.preferences.SettingsPreferences
import java.util.*

class MainActivity : Activity() {


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val settingsPreference = SettingsPreferences.get()
        val stringBuilder = StringBuilder()
        stringBuilder.append("last value firstUse: ").append(settingsPreference.useLanguage).append('\n')
        settingsPreference.useLanguage = Date().toString()

//        text_view.text = stringBuilder.toString()
        // 普通类型保存
        settingsPreference.useLanguage = "zh"
        settingsPreference.lastOpenAppTimeMillis = System.currentTimeMillis()
        // 对象类型保存
        val loginUser = Settings.LoginUser()
        loginUser.username = "username"
        loginUser.password = "password"
        settingsPreference.loginUser = loginUser
        // 对象类型带 @AptField(preferences = true) 注解的保存，相当于把 push相关的放在一个分类
        settingsPreference.push!!.isOpenPush = true


        // 获取
        val useLanguage = settingsPreference.useLanguage
        val loginUser1 = settingsPreference.loginUser
        val openPush = settingsPreference.push!!.isOpenPush

    }

}
